interval/lift/lift.go

```diff
@@ -132,3 +134,66 @@ func StrictBorderCheck(c chain.Chain, i interval.Interval) bool {
 	_, border = chain.TPosToQPos(c, i.GetChromEnd()-1) //interval ranges are open right so we want ChromEnd - 1
 	return border
 }
+
+//everything below is for lifting with AXT
+
+// refCoordToRefIdx is a helper function for LiftCoordinatesWithAxt which identifies the start and end indices in the Axt reference multifa record that corresponds to the interval to be lifted
+func refCoordToRefIdx(a axt.Axt, region interval.Interval) (start, end int) {
+	var stopLoop int
+	for i := range a.RSeq {
+		if stopLoop >= (region.GetChromStart() - (a.RStart - 1)) {
+			break
+		}
+		if a.RSeq[i] != 10 {
+			stopLoop++
+		}
+		start++
+	}
+	end = start
+	stopLoop = 0
+	for i := start; i < len(a.RSeq); i++ {
+		if stopLoop >= (region.GetChromEnd() - region.GetChromStart()) {
+			break
+		}
+		if a.RSeq[i] != 10 {
+			stopLoop++
+		}
+		end++
+	}
+	return start, end
+}
+
+// translateCoord is a helper function for LiftCoordinatesWithAxt which takes the indices of the multifa for the region to be lifted and put them in query coordinates
+func translateCoord(a axt.Axt, start, end int) (newStart, newEnd int) {
```

Comment:
Note to self: this function is finding how "aln pos in query" translates to "ref pos in query", taking gaps into account, kind of like "refPosToAlnPos" for multiFa, but axt can have - strand alignments that get more complicated than in multiFa but allow you to trace back to ref pos in query